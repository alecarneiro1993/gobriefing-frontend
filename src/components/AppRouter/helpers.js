import {
  ROOT_PATH,
  HOME_PATH,
  SIGN_IN_PATH,
  SIGN_UP_PATH,
} from 'utils/constants/routes';
import {
  SIGN_IN_PAGE,
  ROOT_PAGE,
  HOME_PAGE,
  SIGN_UP_PAGE,
} from 'utils/constants/pages';

const PUBLIC_ROUTE = 'PublicRoute';
const PRIVATE_ROUTE = 'PrivateRoute';

export const routes = [
  {
    path: ROOT_PATH,
    exact: true,
    component: ROOT_PAGE,
    route: PUBLIC_ROUTE,
    isRestricted: false,
    key: 'page_root',
  },
  {
    path: SIGN_IN_PATH,
    exact: true,
    component: SIGN_IN_PAGE,
    route: PUBLIC_ROUTE,
    isRestricted: true,
    key: 'page_sign_in',
  },
  {
    path: HOME_PATH,
    exact: true,
    component: HOME_PAGE,
    route: PRIVATE_ROUTE,
    key: 'page_home',
  },
  {
    path: SIGN_UP_PATH,
    exact: true,
    component: SIGN_UP_PAGE,
    route: PUBLIC_ROUTE,
    key: 'page_sign_up',
  },
];
