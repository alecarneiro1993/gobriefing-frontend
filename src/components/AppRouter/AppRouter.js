// @flow

import React from 'react';
import { withCookies } from 'react-cookie';
import { BrowserRouter as Router } from 'react-router-dom';
import { map, isEmpty } from 'lodash';

import { routes } from './helpers';
import * as routeTypes from './components';
import * as pageComponents from '../pages';

type Props = {
  isAuthenticated: boolean,
  cookies: Object
};

type State = {};

class AppRouter extends React.Component<Props, State> {
  render() {
    const { cookies } = this.props;
    const isAuthenticated = !isEmpty(cookies.get('token'));
    return (
      <Router>
        {map(routes, ({ route, key, component, ...routeProps }) => {
          const Page = pageComponents[component];
          const Route = routeTypes[route];
          return (
            <Route
              {...routeProps}
              cookies={cookies}
              key={key}
              component={Page}
              isAuthenticated={isAuthenticated}
            />
          );
        })}
      </Router>
    );
  }
}
export default withCookies(AppRouter);
