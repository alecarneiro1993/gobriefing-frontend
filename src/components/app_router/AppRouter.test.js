import React from "react";
import { shallow } from "enzyme";

import AppRouter from ".";

describe("AppRouter - the routing container component", () => {
  it("renders without crashing", () => {
    const component = shallow(<AppRouter />);
    expect(component).toMatchSnapshot();
  });
});
