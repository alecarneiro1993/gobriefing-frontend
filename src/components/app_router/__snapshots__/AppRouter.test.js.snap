// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter - the routing container component renders without crashing 1`] = `
<BrowserRouter>
  <PublicRoute
    component={[Function]}
    exact={true}
    isAuthenticated={true}
    isRestricted={false}
    key="page_root"
    path="/"
  />
  <PublicRoute
    component={[Function]}
    exact={true}
    isAuthenticated={true}
    isRestricted={true}
    key="page_sign_in"
    path="/sign_in"
  />
  <PrivateRoute
    component={[Function]}
    exact={true}
    isAuthenticated={true}
    key="page_home"
    path="/home"
  />
</BrowserRouter>
`;
